com.hazelcast.client.txn.ClientXATest.testRecovery [API-1657]

_master_ (commit 7f647f2000383001de9675507a0853b2c95e46c3)

Failed on Sonar build (Oracle JDK 11): https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-sonar/1345/testReport/com.hazelcast.client.txn/ClientXATest/testRecovery/

<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: expected:<value> but was:<null>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:120)
	at org.junit.Assert.assertEquals(Assert.java:146)
	at com.hazelcast.client.txn.ClientXATest.testRecovery(ClientXATest.java:196)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

</details>

<details><summary>Standard output:</summary>

```
00:15:38,640  INFO |testRecovery| - [AbstractConfigLocator] testRecovery - Loading 'hazelcast-default.xml' from the classpath.
00:15:38,656  INFO |testRecovery| - [ExternalConfigurationOverride] testRecovery - Detected external configuration entries in environment variables: []
00:15:38,656  WARN |testRecovery| - [ExternalConfigurationOverride] testRecovery - Unrecognized environment variables configuration entries: [hazelcast.test.aws.instance.private.ip]
00:15:38,657  INFO |testRecovery| - [MetricsConfigHelper] testRecovery - [LOCAL] [dev] [5.2.0-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
00:15:38,657  INFO |testRecovery| - [logo] testRecovery - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] 
	+       +  o    o     o     o---o o----o o      o---o     o     o----o o--o--o
	+ +   + +  |    |    / \       /  |      |     /         / \    |         |   
	+ + + + +  o----o   o   o     o   o----o |    o         o   o   o----o    |   
	+ +   + +  |    |  /     \   /    |      |     \       /     \       |    |   
	+       +  o    o o       o o---o o----o o----o o---o o       o o----o    o   
00:15:38,657  INFO |testRecovery| - [system] testRecovery - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
00:15:38,657  INFO |testRecovery| - [system] testRecovery - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Hazelcast Platform 5.2.0-SNAPSHOT (20220919 - 7f647f2) starting at [127.0.0.1]:5701
00:15:38,657  INFO |testRecovery| - [system] testRecovery - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Cluster name: dev
00:15:38,657  INFO |testRecovery| - [system] testRecovery - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Integrity Checker is disabled. Fail-fast on corrupted executables will not be performed. For more information, see the documentation for Integrity Checker.
00:15:38,657  INFO |testRecovery| - [system] testRecovery - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] The Jet engine is disabled.
To enable the Jet engine on the members, do one of the following:
  - Change member config using Java API: config.getJetConfig().setEnabled(true)
  - Change XML/YAML configuration property: Set hazelcast.jet.enabled to true
  - Add system property: -Dhz.jet.enabled=true (for Hazelcast embedded, works only when loading config via Config.load)
  - Add environment variable: HZ_JET_ENABLED=true (recommended when running container image. For Hazelcast embedded, works only when loading config via Config.load)
00:15:38,660  INFO |testRecovery| - [MetricsConfigHelper] testRecovery - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
00:15:38,664  INFO |testRecovery| - [Node] testRecovery - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Using Multicast discovery
00:15:38,664  WARN |testRecovery| - [CPSubsystem] testRecovery - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
00:15:38,667  INFO |testRecovery| - [Diagnostics] testRecovery - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
00:15:38,667  INFO |testRecovery| - [LifecycleService] testRecovery - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] [127.0.0.1]:5701 is STARTING
00:15:41,157  INFO |testRecovery| - [ClusterService] testRecovery - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] 

Members {size:1, ver:1} [
	Member [127.0.0.1]:5701 - 08032ec1-f29f-4c8f-a8b1-e1b6b623f66a this
]

00:15:41,159  INFO |testRecovery| - [LifecycleService] testRecovery - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] [127.0.0.1]:5701 is STARTED
00:15:41,169  INFO |testRecovery| - [AbstractConfigLocator] testRecovery - Loading 'hazelcast-default.xml' from the classpath.
00:15:41,227  INFO |testRecovery| - [ExternalConfigurationOverride] testRecovery - Detected external configuration entries in environment variables: []
00:15:41,227  WARN |testRecovery| - [ExternalConfigurationOverride] testRecovery - Unrecognized environment variables configuration entries: [hazelcast.test.aws.instance.private.ip]
00:15:41,230  INFO |testRecovery| - [MetricsConfigHelper] testRecovery - [LOCAL] [dev] [5.2.0-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
00:15:41,235  INFO |testRecovery| - [logo] testRecovery - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] 
	+       +  o    o     o     o---o o----o o      o---o     o     o----o o--o--o
	+ +   + +  |    |    / \       /  |      |     /         / \    |         |   
	+ + + + +  o----o   o   o     o   o----o |    o         o   o   o----o    |   
	+ +   + +  |    |  /     \   /    |      |     \       /     \       |    |   
	+       +  o    o o       o o---o o----o o----o o---o o       o o----o    o   
00:15:41,236  INFO |testRecovery| - [system] testRecovery - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
00:15:41,236  INFO |testRecovery| - [system] testRecovery - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] Hazelcast Platform 5.2.0-SNAPSHOT (20220919 - 7f647f2) starting at [127.0.0.1]:5702
00:15:41,236  INFO |testRecovery| - [system] testRecovery - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] Cluster name: dev
00:15:41,236  INFO |testRecovery| - [system] testRecovery - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] Integrity Checker is disabled. Fail-fast on corrupted executables will not be performed. For more information, see the documentation for Integrity Checker.
00:15:41,236  INFO |testRecovery| - [system] testRecovery - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] The Jet engine is disabled.
To enable the Jet engine on the members, do one of the following:
  - Change member config using Java API: config.getJetConfig().setEnabled(true)
  - Change XML/YAML configuration property: Set hazelcast.jet.enabled to true
  - Add system property: -Dhz.jet.enabled=true (for Hazelcast embedded, works only when loading config via Config.load)
  - Add environment variable: HZ_JET_ENABLED=true (recommended when running container image. For Hazelcast embedded, works only when loading config via Config.load)
00:15:41,257  INFO |testRecovery| - [MetricsConfigHelper] testRecovery - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
00:15:41,289  INFO |testRecovery| - [Node] testRecovery - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] Using Multicast discovery
00:15:41,289  WARN |testRecovery| - [CPSubsystem] testRecovery - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
00:15:41,308  INFO |testRecovery| - [Diagnostics] testRecovery - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
00:15:41,309  INFO |testRecovery| - [LifecycleService] testRecovery - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] [127.0.0.1]:5702 is STARTING
00:15:41,482  INFO |testRecovery| - [MulticastJoiner] testRecovery - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] Trying to join to discovered node: [127.0.0.1]:5701
00:15:41,487  INFO |testRecovery| - [TcpServerConnection] hz.zen_jennings.IO.thread-in-0 - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:5701 and /127.0.0.1:56461
00:15:41,490  INFO |testRecovery| - [TcpServerConnection] hz.goofy_jennings.IO.thread-in-0 - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] Initialized new cluster connection between /0.0.0.0:56461 and /127.0.0.1:5701
00:15:42,585  INFO |testRecovery| - [ClusterService] hz.zen_jennings.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] 

Members {size:2, ver:2} [
	Member [127.0.0.1]:5701 - 08032ec1-f29f-4c8f-a8b1-e1b6b623f66a this
	Member [127.0.0.1]:5702 - 0a088202-fa4c-4036-a213-fa0221ecabbb
]

00:15:42,585  INFO |testRecovery| - [ClusterService] hz.goofy_jennings.priority-generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] 

Members {size:2, ver:2} [
	Member [127.0.0.1]:5701 - 08032ec1-f29f-4c8f-a8b1-e1b6b623f66a
	Member [127.0.0.1]:5702 - 0a088202-fa4c-4036-a213-fa0221ecabbb this
]

00:15:42,586  INFO |testRecovery| - [LifecycleService] testRecovery - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] [127.0.0.1]:5702 is STARTED
00:15:42,587  INFO |testRecovery| - [AbstractConfigLocator] testRecovery - Loading 'hazelcast-default.xml' from the classpath.
00:15:42,608  INFO |testRecovery| - [ExternalConfigurationOverride] testRecovery - Detected external configuration entries in environment variables: []
00:15:42,608  WARN |testRecovery| - [ExternalConfigurationOverride] testRecovery - Unrecognized environment variables configuration entries: [hazelcast.test.aws.instance.private.ip]
00:15:42,609  INFO |testRecovery| - [MetricsConfigHelper] testRecovery - [LOCAL] [dev] [5.2.0-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
00:15:42,610  INFO |testRecovery| - [logo] testRecovery - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] 
	+       +  o    o     o     o---o o----o o      o---o     o     o----o o--o--o
	+ +   + +  |    |    / \       /  |      |     /         / \    |         |   
	+ + + + +  o----o   o   o     o   o----o |    o         o   o   o----o    |   
	+ +   + +  |    |  /     \   /    |      |     \       /     \       |    |   
	+       +  o    o o       o o---o o----o o----o o---o o       o o----o    o   
00:15:42,610  INFO |testRecovery| - [system] testRecovery - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
00:15:42,610  INFO |testRecovery| - [system] testRecovery - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] Hazelcast Platform 5.2.0-SNAPSHOT (20220919 - 7f647f2) starting at [127.0.0.1]:5703
00:15:42,610  INFO |testRecovery| - [system] testRecovery - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] Cluster name: dev
00:15:42,610  INFO |testRecovery| - [system] testRecovery - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] Integrity Checker is disabled. Fail-fast on corrupted executables will not be performed. For more information, see the documentation for Integrity Checker.
00:15:42,610  INFO |testRecovery| - [system] testRecovery - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] The Jet engine is disabled.
To enable the Jet engine on the members, do one of the following:
  - Change member config using Java API: config.getJetConfig().setEnabled(true)
  - Change XML/YAML configuration property: Set hazelcast.jet.enabled to true
  - Add system property: -Dhz.jet.enabled=true (for Hazelcast embedded, works only when loading config via Config.load)
  - Add environment variable: HZ_JET_ENABLED=true (recommended when running container image. For Hazelcast embedded, works only when loading config via Config.load)
00:15:42,613  INFO |testRecovery| - [MetricsConfigHelper] testRecovery - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
00:15:42,620  INFO |testRecovery| - [Node] testRecovery - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] Using Multicast discovery
00:15:42,620  WARN |testRecovery| - [CPSubsystem] testRecovery - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
00:15:42,625  INFO |testRecovery| - [Diagnostics] testRecovery - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
00:15:42,625  INFO |testRecovery| - [LifecycleService] testRecovery - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] [127.0.0.1]:5703 is STARTING
00:15:42,754  INFO |testRecovery| - [MulticastJoiner] testRecovery - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] Trying to join to discovered node: [127.0.0.1]:5701
00:15:42,755  INFO |testRecovery| - [TcpServerConnection] hz.zen_jennings.IO.thread-in-1 - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:5701 and /127.0.0.1:42651
00:15:42,755  INFO |testRecovery| - [TcpServerConnection] hz.optimistic_jennings.IO.thread-in-0 - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] Initialized new cluster connection between /0.0.0.0:42651 and /127.0.0.1:5701
00:15:43,353  WARN |testRecovery| - [NodeMulticastListener] hz.goofy_jennings.MulticastThread - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] New join request has been received from current master address. The UUID in the join request (62a29dcc-f128-4b24-b442-5f6a09e64dfe) is different from the known master one (08032ec1-f29f-4c8f-a8b1-e1b6b623f66a). Suspecting the master address: [127.0.0.1]:5701
00:15:43,353  WARN |testRecovery| - [MembershipManager] hz.goofy_jennings.MulticastThread - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] Member [127.0.0.1]:5701 - 08032ec1-f29f-4c8f-a8b1-e1b6b623f66a is suspected to be dead for reason: New join request has been received from current master address. The UUID in the join request (62a29dcc-f128-4b24-b442-5f6a09e64dfe) is different from the known master one (08032ec1-f29f-4c8f-a8b1-e1b6b623f66a). Suspecting the master address: [127.0.0.1]:5701
00:15:43,353  INFO |testRecovery| - [MembershipManager] hz.goofy_jennings.MulticastThread - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] Starting mastership claim process...
00:15:43,353  INFO |testRecovery| - [MembershipManager] hz.goofy_jennings.MulticastThread - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] Local MembersView{version=2, members=[MemberInfo{address=[127.0.0.1]:5701, uuid=08032ec1-f29f-4c8f-a8b1-e1b6b623f66a, cpMemberUUID=null, liteMember=false, memberListJoinVersion=1}, MemberInfo{address=[127.0.0.1]:5702, uuid=0a088202-fa4c-4036-a213-fa0221ecabbb, cpMemberUUID=null, liteMember=false, memberListJoinVersion=2}]} with suspected members: [Member [127.0.0.1]:5701 - 08032ec1-f29f-4c8f-a8b1-e1b6b623f66a] and initial addresses to ask: []
00:15:43,354  INFO |testRecovery| - [TcpServerConnection] hz.goofy_jennings.cached.thread-2 - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] Connection[id=1, /0.0.0.0:56461->/127.0.0.1:5701, qualifier=null, endpoint=[127.0.0.1]:5701, remoteUuid=08032ec1-f29f-4c8f-a8b1-e1b6b623f66a, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: Member left event received from master
00:15:43,354  INFO |testRecovery| - [TcpServerConnection] hz.zen_jennings.IO.thread-in-0 - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Connection[id=1, /127.0.0.1:5701->/127.0.0.1:56461, qualifier=null, endpoint=[127.0.0.1]:5702, remoteUuid=0a088202-fa4c-4036-a213-fa0221ecabbb, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: Connection closed by the other side
00:15:43,354  INFO |testRecovery| - [ClusterService] hz.goofy_jennings.cached.thread-2 - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] 

Members {size:1, ver:3} [
	Member [127.0.0.1]:5702 - 0a088202-fa4c-4036-a213-fa0221ecabbb this
]

00:15:43,354  INFO |testRecovery| - [TransactionManagerService] hz.goofy_jennings.cached.thread-3 - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] Committing/rolling-back live transactions of [127.0.0.1]:5701, UUID: 08032ec1-f29f-4c8f-a8b1-e1b6b623f66a
00:15:43,354  INFO |testRecovery| - [MembershipManager] hz.goofy_jennings.cached.thread-2 - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] Mastership is claimed with: MembersView{version=3, members=[MemberInfo{address=[127.0.0.1]:5702, uuid=0a088202-fa4c-4036-a213-fa0221ecabbb, cpMemberUUID=null, liteMember=false, memberListJoinVersion=2}]}
00:15:43,355  INFO |testRecovery| - [TcpServerConnector] hz.zen_jennings.cached.thread-3 - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Connecting to /127.0.0.1:5702, timeout: 10000, bind-any: true
00:15:43,355  INFO |testRecovery| - [TcpServerConnection] hz.goofy_jennings.IO.thread-in-1 - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:5702 and /127.0.0.1:40789
00:15:43,356  INFO |testRecovery| - [TcpServerConnection] hz.zen_jennings.IO.thread-in-2 - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Initialized new cluster connection between /0.0.0.0:40789 and /127.0.0.1:5702
00:15:43,604  INFO |testRecovery| - [InternalPartitionService] hz.goofy_jennings.migration - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] Fetching partition tables from cluster to determine the most recent one... Local stamp: 0
00:15:43,604  INFO |testRecovery| - [InternalPartitionService] hz.goofy_jennings.migration - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] Most recent partition table is determined.
00:15:43,667  INFO |testRecovery| - [ExplicitSuspicionOp] hz.zen_jennings.generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Received suspicion request from: [127.0.0.1]:5702
00:15:43,667  INFO |testRecovery| - [TcpServerConnection] hz.zen_jennings.generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Connection[id=3, /0.0.0.0:40789->/127.0.0.1:5702, qualifier=null, endpoint=[127.0.0.1]:5702, remoteUuid=0a088202-fa4c-4036-a213-fa0221ecabbb, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: explicit suspicion
00:15:43,667  INFO |testRecovery| - [TcpServerConnection] hz.goofy_jennings.IO.thread-in-1 - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] Connection[id=2, /127.0.0.1:5702->/127.0.0.1:40789, qualifier=null, endpoint=[127.0.0.1]:5701, remoteUuid=08032ec1-f29f-4c8f-a8b1-e1b6b623f66a, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: Connection closed by the other side
00:15:43,667  INFO |testRecovery| - [MembershipManager] hz.zen_jennings.generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Removing Member [127.0.0.1]:5702 - 0a088202-fa4c-4036-a213-fa0221ecabbb
00:15:43,668  INFO |testRecovery| - [ClusterService] hz.zen_jennings.generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] 

Members {size:1, ver:3} [
	Member [127.0.0.1]:5701 - 08032ec1-f29f-4c8f-a8b1-e1b6b623f66a this
]

00:15:43,668  INFO |testRecovery| - [TransactionManagerService] hz.zen_jennings.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Committing/rolling-back live transactions of [127.0.0.1]:5702, UUID: 0a088202-fa4c-4036-a213-fa0221ecabbb
00:15:43,856  INFO |testRecovery| - [ClusterService] hz.zen_jennings.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] 

Members {size:2, ver:4} [
	Member [127.0.0.1]:5701 - 08032ec1-f29f-4c8f-a8b1-e1b6b623f66a this
	Member [127.0.0.1]:5703 - e6780df9-8794-49c0-b06a-551285deb874
]

00:15:43,857  INFO |testRecovery| - [ClusterService] hz.optimistic_jennings.priority-generic-operation.thread-0 - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] 

Members {size:2, ver:4} [
	Member [127.0.0.1]:5701 - 08032ec1-f29f-4c8f-a8b1-e1b6b623f66a
	Member [127.0.0.1]:5703 - e6780df9-8794-49c0-b06a-551285deb874 this
]

00:15:43,857  WARN |testRecovery| - [Node] testRecovery - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] Config seed port is 5701 and cluster size is 2. Some of the ports seem occupied!
00:15:43,857  INFO |testRecovery| - [LifecycleService] testRecovery - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] [127.0.0.1]:5703 is STARTED
00:15:43,858  INFO |testRecovery| - [PartitionStateManager] testRecovery - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Initializing cluster partition table arrangement...
00:15:44,593  WARN |testRecovery| - [NodeMulticastListener] hz.optimistic_jennings.MulticastThread - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] New join request has been received from current master address. The UUID in the join request (62a29dcc-f128-4b24-b442-5f6a09e64dfe) is different from the known master one (08032ec1-f29f-4c8f-a8b1-e1b6b623f66a). Suspecting the master address: [127.0.0.1]:5701
00:15:44,593  WARN |testRecovery| - [MembershipManager] hz.optimistic_jennings.MulticastThread - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] Member [127.0.0.1]:5701 - 08032ec1-f29f-4c8f-a8b1-e1b6b623f66a is suspected to be dead for reason: New join request has been received from current master address. The UUID in the join request (62a29dcc-f128-4b24-b442-5f6a09e64dfe) is different from the known master one (08032ec1-f29f-4c8f-a8b1-e1b6b623f66a). Suspecting the master address: [127.0.0.1]:5701
00:15:44,593  INFO |testRecovery| - [MembershipManager] hz.optimistic_jennings.MulticastThread - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] Starting mastership claim process...
00:15:44,593  INFO |testRecovery| - [MembershipManager] hz.optimistic_jennings.MulticastThread - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] Local MembersView{version=4, members=[MemberInfo{address=[127.0.0.1]:5701, uuid=08032ec1-f29f-4c8f-a8b1-e1b6b623f66a, cpMemberUUID=null, liteMember=false, memberListJoinVersion=1}, MemberInfo{address=[127.0.0.1]:5703, uuid=e6780df9-8794-49c0-b06a-551285deb874, cpMemberUUID=null, liteMember=false, memberListJoinVersion=4}]} with suspected members: [Member [127.0.0.1]:5701 - 08032ec1-f29f-4c8f-a8b1-e1b6b623f66a] and initial addresses to ask: []
00:15:44,593  INFO |testRecovery| - [TcpServerConnection] hz.optimistic_jennings.cached.thread-6 - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] Connection[id=1, /0.0.0.0:42651->/127.0.0.1:5701, qualifier=null, endpoint=[127.0.0.1]:5701, remoteUuid=08032ec1-f29f-4c8f-a8b1-e1b6b623f66a, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: Member left event received from master
00:15:44,594  INFO |testRecovery| - [TcpServerConnection] hz.zen_jennings.IO.thread-in-1 - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Connection[id=2, /127.0.0.1:5701->/127.0.0.1:42651, qualifier=null, endpoint=[127.0.0.1]:5703, remoteUuid=e6780df9-8794-49c0-b06a-551285deb874, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: Connection closed by the other side
00:15:44,594  INFO |testRecovery| - [PartitionStateManager] hz.optimistic_jennings.cached.thread-6 - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] Storing snapshot of partition assignments while removing UUID 08032ec1-f29f-4c8f-a8b1-e1b6b623f66a
00:15:44,594  INFO |testRecovery| - [ClusterService] hz.optimistic_jennings.cached.thread-6 - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] 

Members {size:1, ver:5} [
	Member [127.0.0.1]:5703 - e6780df9-8794-49c0-b06a-551285deb874 this
]

00:15:44,594  INFO |testRecovery| - [TransactionManagerService] hz.optimistic_jennings.cached.thread-5 - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] Committing/rolling-back live transactions of [127.0.0.1]:5701, UUID: 08032ec1-f29f-4c8f-a8b1-e1b6b623f66a
00:15:44,594  INFO |testRecovery| - [MembershipManager] hz.optimistic_jennings.cached.thread-6 - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] Mastership is claimed with: MembersView{version=5, members=[MemberInfo{address=[127.0.0.1]:5703, uuid=e6780df9-8794-49c0-b06a-551285deb874, cpMemberUUID=null, liteMember=false, memberListJoinVersion=4}]}
00:15:44,594  INFO |testRecovery| - [TcpServerConnector] hz.zen_jennings.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Connecting to /127.0.0.1:5703, timeout: 10000, bind-any: true
00:15:44,595  INFO |testRecovery| - [TcpServerConnection] hz.optimistic_jennings.IO.thread-in-1 - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:5703 and /127.0.0.1:50857
00:15:44,595  INFO |testRecovery| - [TcpServerConnection] hz.zen_jennings.IO.thread-in-0 - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Initialized new cluster connection between /0.0.0.0:50857 and /127.0.0.1:5703
00:15:44,844  INFO |testRecovery| - [InternalPartitionService] hz.optimistic_jennings.migration - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] Fetching partition tables from cluster to determine the most recent one... Local stamp: 8291625901299219103
00:15:44,844  INFO |testRecovery| - [InternalPartitionService] hz.optimistic_jennings.migration - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] Most recent partition table is determined.
00:15:44,844  INFO |testRecovery| - [InternalPartitionService] hz.optimistic_jennings.migration - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] Applying the most recent of partition state...
00:15:44,844  WARN |testRecovery| - [InternalPartitionService] hz.optimistic_jennings.migration - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] Following unknown addresses are found in partition table sent from master[[127.0.0.1]:5703]. (Probably they have recently joined or left the cluster.) {
	[127.0.0.1]:5701 - 08032ec1-f29f-4c8f-a8b1-e1b6b623f66a
}
00:15:44,879  INFO |testRecovery| - [MigrationManager] hz.optimistic_jennings.migration - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] Partition balance is ok, no need to repartition.
00:15:48,666  INFO |testRecovery| - [ExplicitSuspicionOp] hz.zen_jennings.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Received suspicion request from: [127.0.0.1]:5703
00:15:48,667  INFO |testRecovery| - [TcpServerConnection] hz.zen_jennings.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Connection[id=4, /0.0.0.0:50857->/127.0.0.1:5703, qualifier=null, endpoint=[127.0.0.1]:5703, remoteUuid=e6780df9-8794-49c0-b06a-551285deb874, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: explicit suspicion
00:15:48,667  INFO |testRecovery| - [TcpServerConnection] hz.optimistic_jennings.IO.thread-in-1 - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] Connection[id=2, /127.0.0.1:5703->/127.0.0.1:50857, qualifier=null, endpoint=[127.0.0.1]:5701, remoteUuid=08032ec1-f29f-4c8f-a8b1-e1b6b623f66a, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: Connection closed by the other side
00:15:48,667  INFO |testRecovery| - [MembershipManager] hz.zen_jennings.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Removing Member [127.0.0.1]:5703 - e6780df9-8794-49c0-b06a-551285deb874
00:15:48,667  INFO |testRecovery| - [PartitionStateManager] hz.zen_jennings.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Storing snapshot of partition assignments while removing UUID e6780df9-8794-49c0-b06a-551285deb874
00:15:48,667  INFO |testRecovery| - [ClusterService] hz.zen_jennings.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] 

Members {size:1, ver:5} [
	Member [127.0.0.1]:5701 - 08032ec1-f29f-4c8f-a8b1-e1b6b623f66a this
]

00:15:48,667  INFO |testRecovery| - [TransactionManagerService] hz.zen_jennings.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Committing/rolling-back live transactions of [127.0.0.1]:5703, UUID: e6780df9-8794-49c0-b06a-551285deb874
00:15:48,679  INFO |testRecovery| - [LifecycleService] testRecovery - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] [127.0.0.1]:5701 is SHUTTING_DOWN
00:15:48,688  INFO |testRecovery| - [MigrationManager] hz.zen_jennings.migration - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Partition balance is ok, no need to repartition.
00:15:49,679  INFO |testRecovery| - [MigrationManager] hz.zen_jennings.cached.thread-2 - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Shutdown request of Member [127.0.0.1]:5701 - 08032ec1-f29f-4c8f-a8b1-e1b6b623f66a this is handled
00:15:49,681  INFO |testRecovery| - [Node] testRecovery - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Shutting down multicast service...
00:15:49,691  INFO |testRecovery| - [Node] testRecovery - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Shutting down connection manager...
00:15:49,698  INFO |testRecovery| - [Node] testRecovery - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Shutting down node engine...
00:15:49,729  INFO |testRecovery| - [NodeExtension] testRecovery - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Destroying node NodeExtension.
00:15:49,730  INFO |testRecovery| - [Node] testRecovery - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] Hazelcast Shutdown is completed in 1051 ms.
00:15:49,730  INFO |testRecovery| - [LifecycleService] testRecovery - [127.0.0.1]:5701 [dev] [5.2.0-SNAPSHOT] [127.0.0.1]:5701 is SHUTDOWN
00:15:49,741  INFO |testRecovery| - [AbstractConfigLocator] testRecovery - Loading 'hazelcast-client-default.xml' from the classpath.
00:15:49,797  INFO |testRecovery| - [ClientInvocationService] testRecovery - hz.client_2880 [dev] [5.2.0-SNAPSHOT] Running with 2 response threads, dynamic=true
00:15:49,802  INFO |testRecovery| - [LifecycleService] testRecovery - hz.client_2880 [dev] [5.2.0-SNAPSHOT] HazelcastClient 5.2.0-SNAPSHOT (20220919 - 7f647f2) is STARTING
00:15:49,803  INFO |testRecovery| - [LifecycleService] testRecovery - hz.client_2880 [dev] [5.2.0-SNAPSHOT] HazelcastClient 5.2.0-SNAPSHOT (20220919 - 7f647f2) is STARTED
00:15:49,810  INFO |testRecovery| - [ClientConnectionManager] testRecovery - hz.client_2880 [dev] [5.2.0-SNAPSHOT] Trying to connect to cluster: dev
00:15:49,811  INFO |testRecovery| - [ClientConnectionManager] testRecovery - hz.client_2880 [dev] [5.2.0-SNAPSHOT] Trying to connect to [127.0.0.1]:5701
00:15:49,814  WARN |testRecovery| - [ClientConnectionManager] testRecovery - hz.client_2880 [dev] [5.2.0-SNAPSHOT] Exception during initial connection to [127.0.0.1]:5701: com.hazelcast.core.HazelcastException: java.io.IOException: Connection refused to address /127.0.0.1:5701
00:15:49,814  INFO |testRecovery| - [ClientConnectionManager] testRecovery - hz.client_2880 [dev] [5.2.0-SNAPSHOT] Trying to connect to [127.0.0.1]:5702
00:15:49,820 DEBUG |testRecovery| - [AuthenticationMessageTask] hz.goofy_jennings.priority-generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] Processing authentication with clientUuid 8055e6a7-fc18-451b-a115-8def1f2a050f and clientName hz.client_2880
00:15:49,821  INFO |testRecovery| - [AuthenticationMessageTask] hz.goofy_jennings.priority-generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] Received auth from Connection[id=3, /127.0.0.1:5702->/127.0.0.1:42959, qualifier=null, endpoint=[127.0.0.1]:42959, remoteUuid=8055e6a7-fc18-451b-a115-8def1f2a050f, alive=true, connectionType=JVM, planeIndex=-1], successfully authenticated, clientUuid: 8055e6a7-fc18-451b-a115-8def1f2a050f, client name: hz.client_2880, client version: 5.2.0-SNAPSHOT
00:15:49,822  INFO |testRecovery| - [LifecycleService] testRecovery - hz.client_2880 [dev] [5.2.0-SNAPSHOT] HazelcastClient 5.2.0-SNAPSHOT (20220919 - 7f647f2) is CLIENT_CONNECTED
00:15:49,822  INFO |testRecovery| - [ClientConnectionManager] testRecovery - hz.client_2880 [dev] [5.2.0-SNAPSHOT] Authenticated with server [127.0.0.1]:5702:0a088202-fa4c-4036-a213-fa0221ecabbb, server version: 5.2.0-SNAPSHOT, local address: /127.0.0.1:42959
00:15:49,824  INFO |testRecovery| - [Diagnostics] testRecovery - hz.client_2880 [dev] [5.2.0-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
00:15:49,824 TRACE |testRecovery| - [ClientListenerService] hz.client_2880.internal-1 - hz.client_2880 [dev] [5.2.0-SNAPSHOT] Register attempt of ClusterViewListenerHandler to ClientConnection{alive=true, connectionId=1, channel=NioChannel{/127.0.0.1:42959->/127.0.0.1:5702}, remoteAddress=[127.0.0.1]:5702, lastReadTime=2022-09-20 00:15:49.822, lastWriteTime=2022-09-20 00:15:49.819, closedTime=never, connected server version=null}
00:15:49,829  INFO |testRecovery| - [ClientClusterService] hz.client_2880.event-50243 - hz.client_2880 [dev] [5.2.0-SNAPSHOT] 

Members [1] {
	Member [127.0.0.1]:5702 - 0a088202-fa4c-4036-a213-fa0221ecabbb
}

00:15:49,829 TRACE |testRecovery| - [ClientListenerService] ForkJoinPool.commonPool-worker-15 - hz.client_2880 [dev] [5.2.0-SNAPSHOT] Registered ClusterViewListenerHandler to ClientConnection{alive=true, connectionId=1, channel=NioChannel{/127.0.0.1:42959->/127.0.0.1:5702}, remoteAddress=[127.0.0.1]:5702, lastReadTime=2022-09-20 00:15:49.828, lastWriteTime=2022-09-20 00:15:49.824, closedTime=never, connected server version=null}
00:15:49,835 TRACE |testRecovery| - [ClientListenerService] hz.client_2880.eventRegistration- - hz.client_2880 [dev] [5.2.0-SNAPSHOT] Register attempt of ClientListenerRegistration{codec=com.hazelcast.client.impl.spi.impl.ClientInvocationServiceImpl$1@23834ad3, handler=com.hazelcast.client.impl.spi.impl.ClientInvocationServiceImpl$BackupEventHandler@64fae79b} to ClientConnection{alive=true, connectionId=1, channel=NioChannel{/127.0.0.1:42959->/127.0.0.1:5702}, remoteAddress=[127.0.0.1]:5702, lastReadTime=2022-09-20 00:15:49.828, lastWriteTime=2022-09-20 00:15:49.824, closedTime=never, connected server version=null}
00:15:49,839 TRACE |testRecovery| - [ClientListenerService] hz.client_2880.eventRegistration- - hz.client_2880 [dev] [5.2.0-SNAPSHOT] Registered ClientListenerRegistration{codec=com.hazelcast.client.impl.spi.impl.ClientInvocationServiceImpl$1@23834ad3, handler=com.hazelcast.client.impl.spi.impl.ClientInvocationServiceImpl$BackupEventHandler@64fae79b} to ClientConnection{alive=true, connectionId=1, channel=NioChannel{/127.0.0.1:42959->/127.0.0.1:5702}, remoteAddress=[127.0.0.1]:5702, lastReadTime=2022-09-20 00:15:49.839, lastWriteTime=2022-09-20 00:15:49.836, closedTime=never, connected server version=null}
00:15:49,841  INFO |testRecovery| - [ClientStatisticsService] testRecovery - Client statistics is enabled with period 5 seconds.
00:15:49,854  INFO |testRecovery| - [PartitionStateManager] hz.goofy_jennings.partition-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] Initializing cluster partition table arrangement...
00:15:50,343  INFO |testRecovery| - [LifecycleService] main - hz.client_2880 [dev] [5.2.0-SNAPSHOT] HazelcastClient 5.2.0-SNAPSHOT (20220919 - 7f647f2) is SHUTTING_DOWN
00:15:50,344  INFO |testRecovery| - [ClientConnectionManager] main - hz.client_2880 [dev] [5.2.0-SNAPSHOT] Removed connection to endpoint: [127.0.0.1]:5702:0a088202-fa4c-4036-a213-fa0221ecabbb, connection: ClientConnection{alive=false, connectionId=1, channel=NioChannel{/127.0.0.1:42959->/127.0.0.1:5702}, remoteAddress=[127.0.0.1]:5702, lastReadTime=2022-09-20 00:15:50.305, lastWriteTime=2022-09-20 00:15:49.856, closedTime=2022-09-20 00:15:50.344, connected server version=null}
00:15:50,344  INFO |testRecovery| - [LifecycleService] main - hz.client_2880 [dev] [5.2.0-SNAPSHOT] HazelcastClient 5.2.0-SNAPSHOT (20220919 - 7f647f2) is CLIENT_DISCONNECTED
00:15:50,344  INFO |testRecovery| - [TcpServerConnection] hz.goofy_jennings.IO.thread-in-2 - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] Connection[id=3, /127.0.0.1:5702->/127.0.0.1:42959, qualifier=null, endpoint=[127.0.0.1]:42959, remoteUuid=8055e6a7-fc18-451b-a115-8def1f2a050f, alive=false, connectionType=JVM, planeIndex=-1] closed. Reason: Connection closed by the other side
00:15:50,344  INFO |testRecovery| - [ClientEndpointManager] hz.goofy_jennings.event-50237 - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] Destroying ClientEndpoint{connection=Connection[id=3, /127.0.0.1:5702->/127.0.0.1:42959, qualifier=null, endpoint=[127.0.0.1]:42959, remoteUuid=8055e6a7-fc18-451b-a115-8def1f2a050f, alive=false, connectionType=JVM, planeIndex=-1], clientUuid=8055e6a7-fc18-451b-a115-8def1f2a050f, clientName=hz.client_2880, authenticated=true, clientVersion=5.2.0-SNAPSHOT, creationTime=1663632949820, latest clientAttributes=lastStatisticsCollectionTime=1663632949840,enterprise=false,clientType=JVM,clientVersion=5.2.0-SNAPSHOT,clusterConnectionTimestamp=1663632949816,clientAddress=127.0.0.1,clientName=hz.client_2880,credentials.principal=null,os.committedVirtualMemorySize=25830682624,os.freePhysicalMemorySize=335991840768,os.freeSwapSpaceSize=8587833344,os.maxFileDescriptorCount=1048576,os.openFileDescriptorCount=365,os.processCpuTime=19435150000000,os.systemLoadAverage=7.13,os.totalPhysicalMemorySize=811486535680,os.totalSwapSpaceSize=8589930496,runtime.availableProcessors=8,runtime.freeMemory=170177832,runtime.maxMemory=2147483648,runtime.totalMemory=748683264,runtime.uptime=8758479,runtime.usedMemory=578505432, labels=[]}
00:15:50,348  INFO |testRecovery| - [LifecycleService] main - hz.client_2880 [dev] [5.2.0-SNAPSHOT] HazelcastClient 5.2.0-SNAPSHOT (20220919 - 7f647f2) is SHUTDOWN
00:15:50,350  INFO |testRecovery| - [LifecycleService] main - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] [127.0.0.1]:5702 is SHUTTING_DOWN
00:15:50,350  INFO |testRecovery| - [MigrationManager] hz.goofy_jennings.cached.thread-4 - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] Shutdown request of Member [127.0.0.1]:5702 - 0a088202-fa4c-4036-a213-fa0221ecabbb this is handled
00:15:50,351  INFO |testRecovery| - [Node] main - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] Shutting down multicast service...
00:15:50,351  INFO |testRecovery| - [Node] main - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] Shutting down connection manager...
00:15:50,355  INFO |testRecovery| - [Node] main - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] Shutting down node engine...
00:15:50,359  INFO |testRecovery| - [NodeExtension] main - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] Destroying node NodeExtension.
00:15:50,359  INFO |testRecovery| - [Node] main - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] Hazelcast Shutdown is completed in 9 ms.
00:15:50,359  INFO |testRecovery| - [LifecycleService] main - [127.0.0.1]:5702 [dev] [5.2.0-SNAPSHOT] [127.0.0.1]:5702 is SHUTDOWN
00:15:50,359  INFO |testRecovery| - [LifecycleService] main - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] [127.0.0.1]:5703 is SHUTTING_DOWN
00:15:50,359  INFO |testRecovery| - [MigrationManager] hz.optimistic_jennings.cached.thread-3 - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] Shutdown request of Member [127.0.0.1]:5703 - e6780df9-8794-49c0-b06a-551285deb874 this is handled
00:15:50,359  INFO |testRecovery| - [Node] main - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] Shutting down multicast service...
00:15:50,359  INFO |testRecovery| - [Node] main - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] Shutting down connection manager...
00:15:50,360  INFO |testRecovery| - [Node] main - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] Shutting down node engine...
00:15:50,363  INFO |testRecovery| - [NodeExtension] main - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] Destroying node NodeExtension.
00:15:50,363  INFO |testRecovery| - [Node] main - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] Hazelcast Shutdown is completed in 4 ms.
00:15:50,363  INFO |testRecovery| - [LifecycleService] main - [127.0.0.1]:5703 [dev] [5.2.0-SNAPSHOT] [127.0.0.1]:5703 is SHUTDOWN
```

</details>

--- a/hazelcast/src/test/java/com/hazelcast/client/txn/ClientXATest.java
+++ b/hazelcast/src/test/java/com/hazelcast/client/txn/ClientXATest.java
@@ -48,6 +48,7 @@ import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 
+import static com.hazelcast.test.HazelcastTestSupport.assertClusterSize;
 import static com.hazelcast.test.HazelcastTestSupport.assertOpenEventually;
 import static com.hazelcast.test.HazelcastTestSupport.randomString;
 import static com.hazelcast.test.HazelcastTestSupport.sleepAtLeastSeconds;
@@ -175,8 +176,10 @@ public class ClientXATest {
     @Test
     public void testRecovery() throws Exception {
         HazelcastInstance instance = Hazelcast.newHazelcastInstance();
-        Hazelcast.newHazelcastInstance();
-        Hazelcast.newHazelcastInstance();
+        HazelcastInstance instance2 = Hazelcast.newHazelcastInstance();
+        HazelcastInstance instance3 = Hazelcast.newHazelcastInstance();
+
+        assertClusterSize(3, instance, instance2, instance3);
 
         HazelcastXAResource xaResource = instance.getXAResource();
         Xid myXid = new SerializableXID(42, "globalTransactionId".getBytes(), "branchQualifier".getBytes());

--- a/hazelcast/src/test/java/com/hazelcast/client/txn/ClientXATest.java
+++ b/hazelcast/src/test/java/com/hazelcast/client/txn/ClientXATest.java
@@ -17,8 +17,7 @@
 package com.hazelcast.client.txn;
 
 import com.atomikos.icatch.jta.UserTransactionManager;
-import com.hazelcast.client.HazelcastClient;
-import com.hazelcast.core.Hazelcast;
+import com.hazelcast.client.test.TestHazelcastFactory;
 import com.hazelcast.core.HazelcastInstance;
 import com.hazelcast.map.IMap;
 import com.hazelcast.collection.IQueue;
@@ -62,6 +61,7 @@ import static org.junit.Assert.fail;
 @Category(SlowTest.class)
 public class ClientXATest {
 
+    private final TestHazelcastFactory hazelcastFactory = new TestHazelcastFactory();
     static final Random random = new Random(System.currentTimeMillis());
     static final ILogger logger = Logger.getLogger(ClientXATest.class);
 
@@ -91,22 +91,20 @@ public class ClientXATest {
         cleanAtomikosLogs();
         tm = new UserTransactionManager();
         tm.setTransactionTimeout(60);
-        HazelcastClient.shutdownAll();
-        Hazelcast.shutdownAll();
+        hazelcastFactory.terminateAll();
     }
 
     @After
     public void cleanup() {
         tm.close();
         cleanAtomikosLogs();
-        HazelcastClient.shutdownAll();
-        Hazelcast.shutdownAll();
+        hazelcastFactory.terminateAll();
     }
 
     @Test
     public void testRollbackOnTimeout() throws Exception {
-        Hazelcast.newHazelcastInstance();
-        HazelcastInstance client = HazelcastClient.newHazelcastClient();
+        hazelcastFactory.newHazelcastInstance();
+        HazelcastInstance client = hazelcastFactory.newHazelcastClient();
 
         String name = randomString();
         IQueue<Object> queue = client.getQueue(name);
@@ -132,8 +130,8 @@ public class ClientXATest {
 
     @Test
     public void testWhenLockedOutOfTransaction() throws Exception {
-        Hazelcast.newHazelcastInstance();
-        HazelcastInstance client = HazelcastClient.newHazelcastClient();
+        hazelcastFactory.newHazelcastInstance();
+        HazelcastInstance client = hazelcastFactory.newHazelcastClient();
         IMap<Object, Object> map = client.getMap("map");
         map.put("key", "value");
         HazelcastXAResource xaResource = client.getXAResource();
@@ -151,8 +149,8 @@ public class ClientXATest {
 
     @Test
     public void testRollback() throws Exception {
-        Hazelcast.newHazelcastInstance();
-        HazelcastInstance client = HazelcastClient.newHazelcastClient();
+        hazelcastFactory.newHazelcastInstance();
+        HazelcastInstance client = hazelcastFactory.newHazelcastClient();
         HazelcastXAResource xaResource = client.getXAResource();
 
         tm.begin();
@@ -175,9 +173,9 @@ public class ClientXATest {
 
     @Test
     public void testRecovery() throws Exception {
-        HazelcastInstance instance = Hazelcast.newHazelcastInstance();
-        HazelcastInstance instance2 = Hazelcast.newHazelcastInstance();
-        HazelcastInstance instance3 = Hazelcast.newHazelcastInstance();
+        HazelcastInstance instance = hazelcastFactory.newHazelcastInstance();
+        HazelcastInstance instance2 = hazelcastFactory.newHazelcastInstance();
+        HazelcastInstance instance3 = hazelcastFactory.newHazelcastInstance();
 
         assertClusterSize(3, instance, instance2, instance3);
 
@@ -190,7 +188,7 @@ public class ClientXATest {
         xaResource.prepare(myXid);
         instance.shutdown();
 
-        HazelcastInstance client = HazelcastClient.newHazelcastClient();
+        HazelcastInstance client = hazelcastFactory.newHazelcastClient();
         HazelcastXAResource clientXaResource = client.getXAResource();
         Xid[] recovered = clientXaResource.recover(0);
         for (Xid xid : recovered) {
@@ -201,11 +199,11 @@ public class ClientXATest {
 
     @Test
     public void testIsSame() throws Exception {
-        HazelcastInstance instance1 = Hazelcast.newHazelcastInstance();
-        HazelcastInstance instance2 = Hazelcast.newHazelcastInstance();
+        HazelcastInstance instance1 = hazelcastFactory.newHazelcastInstance();
+        HazelcastInstance instance2 = hazelcastFactory.newHazelcastInstance();
         XAResource resource1 = instance1.getXAResource();
         XAResource resource2 = instance2.getXAResource();
-        HazelcastInstance client = HazelcastClient.newHazelcastClient();
+        HazelcastInstance client = hazelcastFactory.newHazelcastClient();
         XAResource clientResource = client.getXAResource();
         assertTrue(clientResource.isSameRM(resource1));
         assertTrue(clientResource.isSameRM(resource2));
@@ -213,8 +211,8 @@ public class ClientXATest {
 
     @Test
     public void testParallel() throws Exception {
-        Hazelcast.newHazelcastInstance();
-        final HazelcastInstance client = HazelcastClient.newHazelcastClient();
+        hazelcastFactory.newHazelcastInstance();
+        final HazelcastInstance client = hazelcastFactory.newHazelcastClient();
         // this is needed due to a racy bug in atomikos
         txn(client);
         int size = 100;
@@ -242,8 +240,8 @@ public class ClientXATest {
 
     @Test
     public void testSequential() throws Exception {
-        Hazelcast.newHazelcastInstance();
-        HazelcastInstance client = HazelcastClient.newHazelcastClient();
+        hazelcastFactory.newHazelcastInstance();
+        HazelcastInstance client = hazelcastFactory.newHazelcastClient();
         int count = 100;
         for (int i = 0; i < count; i++) {
             txn(client);

